lib/browser-pool/caching-pool.js

```diff
@@ -35,7 +36,13 @@ CachingPool.prototype.getBrowser = function(id) {
     }
 
     log('has cached browser %o', browser);
-    return browser.reset().thenResolve(browser);
+    return browser.reset()
+        .fail(function(e) {
+            var reject = q.reject.bind(null, e);
+            return this.freeBrowser(browser)
```

Comment:
we've got it from the cache, so we put it back. There can be some temporary problems with network or else.